vars/publishToMaven.groovy

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+/** Library to sign and deploy opensearch artifacts to a sonatype staging repository.
+ @param Map args = [:] args A map of the following parameters
+ @param args.signingArtifactsPath <required> - Path to yml or artifact file. This could be path to Dir that contains yml file or full path to yml file.
+ @param args.mavenArtifactsPath <required> - The directory containing distribution files to upload to the repository. example: ~/.m2/repository where repository contains /org/opensearch
```

Comment:
Let's mention that these paths have to be local. Not sure how this will work for urls and stuff. Also you forgot about autoPublish: true or false. I believe you need to pass that arg to underlying script
```
 @param args.signingArtifactsPath <required> - Path to yml, an artifact or directory containing a collection of artifacts. This could be path to Dir that contains yml file or full path to yml file.
 @param args.mavenArtifactsPath <required> - The directory containing artifacts to publish to maven repository example: ~/.m2/repository where repository contains /org/opensearch
@param args.autoPublish <optional> - Wether to publish artifacts directly to maven without human intervention. Defaults to false
```

Reply:
That arg was being set up as env variable in the jenkinsfile and then being used as env variable in the shell script. I think it will make more sense to pass it as an argument than using an env variable. Will make the required changes and update.  @gaiksaya